### PR TITLE
timer: refactor to make reactive, update tests and preview

### PR DESCRIPTION
**Pull request by otonashixav**

### CODE REVIEW

**Comment by davedbase** [comment]
This looks pretty good @otonashixav. I believe @thetarnav had his own implementation as well. Are you suggesting this in favour of his or can we attempt to consolidate ideas?

**Comment by otonashixav** [comment]
@davedbase consolidated these:
- Delays can be false to disable the timer. 
- Allow options to be passed to the signal in `createPolled`.

Remaining differences:
- Can't pass an initial value to `createPolled` -- if you're polling a function you probably want only values that come from it. 
- Can't pass options to `createIntervalCounter` -- other than passing a name for the signal, seems unneeded.

**Comment by thetarnav** [comment]
I was vibing with `newTimer` a little bit more... but whatever, this also works.

I agree with the problem of `createPolled` initial value in my implementation. It shouldn't be an actual initial value that get's returned, BUT the api could still include an initial value in the same way the `createMemo` does – for the initial value of the `prev` argument in the calc function. It's useful to change `T | undefined` to just `T`.
The api could be similar to `createMemo`:
```ts
createMemo(calc, initialValue, options);
createPolled(calc, timeout, initialValue, options);
```

And signal options in `createIntervalCounter`... why not? It's just a pass through anyways.

The rest is solid :)

File: packages/timer/README.md [hunk 1]
```diff
@@ -5,28 +5,108 @@
...
+[![stage](https://img.shields.io/endpoint?style=for-the-badge&url=https%3A%2F%2Fraw.githubusercontent.com%2Fsolidjs-community%2Fsolid-primitives%2Fmain%2Fassets%2Fbadges%2Fstage-0.json)](https://github.com/solidjs-community/solid-primitives#contribution-process)
 
-A timer wrapper to handle setTimeout and setInterval.
+Timer primitives related to `setInterval` and `setTimeout`:
+
+- `createTimeout` - setTimeout with an optionally reactive delay.
+- `createInterval` - setInterval with an optionally reactive delay.
+- `createTimeoutLoop` - Like createInterval, except the delay only updates between executions.
+- `createPolled` - Polls a function periodically. Returns an to the latest polled value.
+- `createIntervalCounter` - Creates a counter which increments periodically.
```
Comment (thetarnav) [inline packages/timer/README.md:18]: TOC should have links to their sections.
```suggestion
Timer primitives related to [`setInterval`](https://developer.mozilla.org/en-US/docs/Web/API/setInterval) and [`setTimeout`](https://developer.mozilla.org/en-US/docs/Web/API/setTimeout):

- [`makeTimer`](#makeTimer) - Makes an automatically cleaned up timer.
- [`createTimer`](#createTimer) - [makeTimer](#maketimer), but with a fully reactive delay
- [`createTimeoutLoop`](#createTimeoutLoop) - Like createInterval, except the delay only updates between executions.
- [`createPolled`](#createPolled) - Polls a function periodically. Returns an to the latest polled value.
- [`createIntervalCounter`](#createIntervalCounter) - Creates a counter which increments periodically.
```

File: packages/timer/package.json [hunk 2]
```diff
@@ -1,33 +1,44 @@
 {
   "name": "@solid-primitives/timer",
-  "version": "1.2.0",
-  "description": "Primitive to manage timeout and interval",
+  "version": "1.3.0",
+  "description": "Primitives to manage timeout and interval",
   "author": "David Di Biase <dave.dibiase@gmail.com>",
+  "contributors": [],
```
Comment (thetarnav) [inline packages/timer/package.json:6]: You should add yourself to the "contributors" field. :)

File: packages/timer/README.md [hunk 3]
```diff
@@ -5,28 +5,108 @@
...
-[![size](https://img.shields.io/bundlephobia/minzip/@solid-primitives/timer?style=for-the-badge)](https://bundlephobia.com/package/@solid-primitives/timer)
-[![size](https://img.shields.io/npm/v/@solid-primitives/timer?style=for-the-badge)](https://www.npmjs.com/package/@solid-primitives/timer)
-[![stage](https://img.shields.io/endpoint?style=for-the-badge&url=https%3A%2F%2Fraw.githubusercontent.com%2Fsolidjs-community%2Fsolid-primitives%2Fmain%2Fassets%2Fbadges%2Fstage-3.json)](https://github.com/solidjs-community/solid-primitives#contribution-process)
+[![size](https://img.shields.io/bundlephobia/minzip/@solid-primitives/timer?style=for-the-badge&label=size)](https://bundlephobia.com/package/@solid-primitives/timer)
+[![version](https://img.shields.io/npm/v/@solid-primitives/timer?style=for-the-badge)](https://www.npmjs.com/package/@solid-primitives/timer)
+[![stage](https://img.shields.io/endpoint?style=for-the-badge&url=https%3A%2F%2Fraw.githubusercontent.com%2Fsolidjs-community%2Fsolid-primitives%2Fmain%2Fassets%2Fbadges%2Fstage-0.json)](https://github.com/solidjs-community/solid-primitives#contribution-process)
```
Comment (thetarnav) [inline packages/timer/README.md:10]:
```suggestion
[![stage](https://img.shields.io/endpoint?style=for-the-badge&url=https%3A%2F%2Fraw.githubusercontent.com%2Fsolidjs-community%2Fsolid-primitives%2Fmain%2Fassets%2Fbadges%2Fstage-3.json)](https://github.com/solidjs-community/solid-primitives#contribution-process)
```

**Review by thetarnav — APPROVED** [review]

**Comment by thetarnav** [comment]
Published to `@solid-primitives/timer@1.3.0`

**Pull request merged**